src/main/webapp/dev/js/common/feedbackResponseComments.js

```diff
@@ -562,8 +567,111 @@ function registerResponseCommentCheckboxEvent() {
+                ['recipientindex', 'giverindex', 'qnindex', 'sectionindex']]);
+    clickHandlerMap.set(
+            '.hide-frc-edit-form', [hideResponseCommentEditForm,
+                ['recipientindex', 'giverindex', 'qnindex', 'frcindex', 'sectionindex', 'viewtype']]);
+    clickHandlerMap.set(
+            '.toggle-visib-add-form', [toggleVisibilityAddForm,
+                ['recipientindex', 'giverindex', 'qnindex', 'sectionindex']]);
+    clickHandlerMap.set(
+            '.toggle-visib-edit-form', [toggleVisibilityEditForm,
+                ['recipientindex', 'giverindex', 'qnindex', 'frcindex', 'sectionindex', 'viewtype']]);
+
+    /* eslint-disable no-restricted-syntax */
+    for (const [className, clickHandlerAndParams] of clickHandlerMap) {
+        $(document).on('click', className, (e) => {
+            const ev = $(e.currentTarget);
+            const clickHandler = clickHandlerAndParams[0];
+            const params = clickHandlerAndParams[1].map(paramName => ev.data(paramName));
+            clickHandler(params[0], params[1], params[2], params[3], params[4], params[5]);
+        });
+    }
+    /* eslint-enable no-restricted-syntax */
+}
+
+function registerResponseCommentCheckboxEventForFeedbackPage() {
+    $('body').on('click', 'ul[id^="responseCommentTable"] * input[type=checkbox]', (e) => {
+        const table = $(e.currentTarget).closest('table');
+        const parentDiv = table.parent().attr('id');
+        const responseCommentId = parentDiv.substring('visibility-options-'.length);
+        let visibilityOptions = [];
+        const target = $(e.target);
+        const visibilityOptionsRow = target.closest('tr');
+        if (target.prop('class').includes('answerCheckbox') && !target.prop('checked')) {
+            visibilityOptionsRow.find('input[class*=giverCheckbox]').prop('checked', false);
+            visibilityOptionsRow.find('input[class*=recipientCheckbox]').prop('checked', false);
+        }
+        if ((target.prop('class').includes('giverCheckbox') || target.prop('class').includes('recipientCheckbox'))
+            && target.prop('checked')) {
+            visibilityOptionsRow.find('input[class*=answerCheckbox]').prop('checked', true);
+        }
+
+        table.find('.answerCheckbox:checked').each(function () {
+            visibilityOptions.push($(this).val());
+        });
+        $(`input[name=showresponsecommentsto-${responseCommentId}]`).val(visibilityOptions.join(', '));
+
+        visibilityOptions = [];
+        table.find('.giverCheckbox:checked').each(function () {
+            visibilityOptions.push($(this).val());
+        });
+        $(`input[name=showresponsegiverto-${responseCommentId}]`).val(visibilityOptions.join(', '));
+    });
+}
+
 export {
     enableHoverToDisplayEditOptions,
     registerResponseCommentCheckboxEvent,
     registerResponseCommentsEvent,
+    registerResponseCommentCheckboxEventForFeedbackPage,
+    registerResponseCommentsEventForFeedbackPage,
```

Comment:
What's the difference between the original one and the new one?

Reply:
Original one deals with `form` and clickHandlerMaps for visibility settings. New one deals with `div`.

Reply:
Okay, in that case, as our discussion indicated above, we shall have separate comment row tag for feedback submission and also separate JS code for feedback submission.